### PR TITLE
chore: fix reset button types

### PR DESCRIPTION
**Pull request by SiTaggart**
Description and useful links to discussions / issues / tickets

### Fix the reset variant button types

Conditionally allow style props on a reset button variant

### Improve button types

ButtonProps didn't extend DirectButtonProps. This led to a mismatch between the consumer types and the internal types. For example, to a consumer we didn't restrict tabIndex to 0 and -1 but we did internally.

### Iput and User Dialog removal of ts expect error

No longer needed for the reset button as style props are valid.

## Checklist

- [ ] Work in a "Draft" branch whilst you are getting feedback to reduce Chromatic costs. Once you are ready for approval, convert the PR to "Ready to review"
- [ ] Ensure all `required` checks have passed
- [ ] Icon changes must have a product design review
- [ ] Website content changes must have a product design review
- [ ] At least two engineers have reviewed any code changes
- [ ] At least one engineering lead has reviewed any core package changes or repository infrastructure
- [ ] Website visual regression testing is run by adding `🕵🏻‍♀️ Run website visual regression` label

### CODE REVIEW

**Review by TheSisb — APPROVED** [review]
perfect

File: .changeset/brave-gorillas-begin.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@twilio-paste/user-dialog': patch
+'@twilio-paste/core': patch
+---
+
+Removal of ts annotation to expect an error for reset button style overrides
```
Comment (TheSisb) [inline .changeset/brave-gorillas-begin.md:6]:
```suggestion
[User Dialog] Removal of ts annotation to expect an error for reset button style overrides
```

Comment (TheSisb) [reply .changeset/brave-gorillas-begin.md:6]: Although I wonder if we even need to ship this as a patch since it doesn't affect users really?

**Review by TheSisb — DISMISSED** [review]
Ah I had a feeling a few spots must have been missed in my first approval. I should have dug into that hunch.

File: packages/paste-core/components/button/src/types.ts [hunk 2]
```diff
@@ -34,32 +34,30 @@ export interface ButtonContentsProps {
...
-  target?: string;
-}
+type BaseVariantsButtonProps = {
+  variant?: Omit<ButtonVariants, 'reset'>;
+};
+type ResetVariantButtonProps = BoxStyleProps & {
+  variant?: 'reset';
+};
+
+export type ButtonProps = (BaseVariantsButtonProps | ResetVariantButtonProps) &
```
Comment (TheSisb) [inline packages/paste-core/components/button/src/types.ts:58]: Do we want to add docs for this? An example of using reset button? Or keep it under wraps for now?

**Review by nkrantz — DISMISSED** [review]
suggestion(non-blocking): can you add to [the docs example](https://paste.twilio.design/components/button#reset-button-styles) explaining that our tokens can be used to style reset buttons?

File: .changeset/brave-gorillas-begin.md (hunk 1 above)
Comment (SiTaggart) [reply .changeset/brave-gorillas-begin.md:6]: Probably not, it's just to appease the Danger JS check

**Review by nkrantz — APPROVED** [review]

**Pull request merged**